src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsForm/TransformationsForm.js

```diff
@@ -1,27 +1,41 @@
-import React from 'react';
+import React, {
+  memo,
+  useCallback,
+} from 'react';
 import PropTypes from 'prop-types';
+import { intersectionWith } from 'lodash';
 
 import stripesFinalForm from '@folio/stripes/final-form';
 
 import { TransformationField } from '../TransformationsField';
 
 import css from './TransformationsForm.css';
 
-const TransformationsFormComponent = ({
+const TransformationsFormComponent = memo(({
   autosize,
+  searchResults,
   handleSubmit,
 }) => {
+  const prepareData = useCallback(
```

Comment:
I'm note sure that I got your idea about this property. If we put "searchResults" property, so why do we need to apply intersectionWith or identity again? Maybe, I misunderstand the idea. But  searchResult means the list of search results and we need just to render it. But according to the implementation, we have to apply some handlers again.

Reply:
searchResults are based on the initialProps which does not contain the information from the form such as the values of transformation fields. The [fields](https://github.com/folio-org/ui-data-export/blob/UIDEXP-69/src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js#L75) object contains it.

Reply:
Have discussed the chosen approach and agreed that I will try the context approach as an alternative solution in order to simplify the data flow and thus make the code easier to reason about.